lib/sanbase/blockchain/timescaledb.ex

```
@@ -100,19 +100,24 @@ defmodule Sanbase.Timescaledb do
   @spec timescaledb_execute({iolist() | String.t(), list(argument)}, (any() -> any())) ::
           {:ok, list(any())} | {:error, any()}
   def timescaledb_execute({query, args}, transform_fn) when is_function(transform_fn, 1) do
-    Sanbase.TimescaleRepo.query(query, args)
-    |> case do
-      {:ok, %{rows: rows}} ->
-        result =
-          Enum.map(
-            rows,
-            transform_fn
-          )
-
-        {:ok, result}
-
-      {:error, error} ->
-        {:error, error}
+    try do
+      Sanbase.TimescaleRepo.query(query, args)
+      |> case do
+        {:ok, %{rows: rows}} ->
+          result =
+            Enum.map(
+              rows,
+              transform_fn
+            )
+
+          {:ok, result}
+
+        {:error, error} ->
+          {:error, error}
+      end
+    rescue
+      e ->
+        {:error, "Cannot execute TimescaleDB query. Reason: " <> Exception.message(e)}
     end
```

It is really weird that the repo is throwing an exception instead of returning an error tuple.

---

Depends. The functions without `!` can still raise if the error is not related to the function responsibilities. Not having a database/not being able to connect to db is example for such cases